### PR TITLE
fix: (Platform) Removed dependency on click event for Radio button group

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
#4282 

#### Please provide a brief summary of this pull request.
Previous implementation had following bugs:
1. for vertical radio group, radio button was getting clicked from any horizontal position
2. selection was getting removed without selection of other radio
3. removed code duplication

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- N/A [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/form/radio-group/radio-group.component.ts [hunk 1]
```diff
@@ -205,52 +205,38 @@ export class RadioGroupComponent extends CollectionBaseInput implements AfterVie
...
-                }
-                button.click.pipe(takeUntil(this.destroy$)).subscribe((ev) => this._selectedValueChanged(ev));
-            });
-            // accessibility requirement
-            if (!this._selected && this.contentRadioButtons && firstEnabledButtonIndex > -1) {
-                this.contentRadioButtons.toArray()[firstEnabledButtonIndex].setTabIndex(0);
-            }
-            this.onChange(this.value);
-        }
+        this._initialSetup(this.contentRadioButtons);
```
Comment (olshansky) [inline libs/platform/src/lib/components/form/radio-group/radio-group.component.ts:208]: Is it really necessary to have a method just to call another method? 
You could remove `_initContentRadioButtons` and move `this._initialSetup(this.contentRadioButtons);` to `ngAfterViewInit` hook

File: libs/platform/src/lib/components/form/radio-group/radio-group.component.ts [hunk 2]
```diff
@@ -205,52 +205,38 @@ export class RadioGroupComponent extends CollectionBaseInput implements AfterVie
...
     }
 
     /**
      * Select radio button with provided value
      */
     private _initViewRadioButtons(): void {
-        if (this.viewRadioButtons && this.viewRadioButtons.length > 0) {
+        this._initialSetup(this.viewRadioButtons);
+    }
```
Comment (olshansky) [inline libs/platform/src/lib/components/form/radio-group/radio-group.component.ts:216]: The same as above

File: libs/platform/src/lib/components/form/radio-group/radio-group.component.ts (hunk 1 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/form/radio-group/radio-group.component.ts]: fixed

File: libs/platform/src/lib/components/form/radio-group/radio-group.component.ts (hunk 2 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/form/radio-group/radio-group.component.ts]: fixed

**Review by olshansky — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Pull request merged**